src/Shared/AssemblyUtilities.cs

```diff
@@ -55,22 +54,41 @@ public static Type GetTypeInfo(this Type t)
 
         public static AssemblyName CloneIfPossible(this AssemblyName assemblyNameToClone)
         {
-#if FEATURE_ASSEMBLYNAME_CLONE
+#if CLR2COMPATIBILITY
             return (AssemblyName) assemblyNameToClone.Clone();
 #else
 
-            Initialize();
-
-            if (s_assemblyNameCloneMethod == null)
-            {
-                return new AssemblyName(assemblyNameToClone.FullName);
-            }
+            // NOTE: In large projects, this is called a lot. Avoid calling AssemblyName.Clone
+            // because it clones the Version property (which is immutable) and the PublicKey property
+            // and the PublicKeyToken property.
+            //
+            // While the array themselves are mutable - throughout MSBuild they are only ever
+            // read from.
+            AssemblyName name = new AssemblyName();
+            name.Name = assemblyNameToClone.Name;
+            name.SetPublicKey(assemblyNameToClone.GetPublicKey());
+            name.SetPublicKeyToken(assemblyNameToClone.GetPublicKeyToken());
+            name.Version = assemblyNameToClone.Version;
+            name.Flags = assemblyNameToClone.Flags;
+            name.ContentType = assemblyNameToClone.ContentType;
+            name.ProcessorArchitecture = assemblyNameToClone.ProcessorArchitecture;
+
+#if FEATURE_ASSEMBLYNAME_CULTUREINFO
```

Comment:
Should this section be restricted to setting `name.CultureInfo` only?

Reply:
I was thinking about that. All the members under the flag are not supported on .net core. Since we already have a mess of API flags, I didn't want to increase the proliferation. I'll switch it with `RUNTIME_TYPE_NETCORE` instead.